app/models/user.rb

```diff
@@ -116,6 +110,17 @@ def authentication_token_expired?
     auth_token_expiry.nil? || auth_token_expiry <= Time.zone.now
   end
 
+  #
+  # Returns authentication of the user
+  #
+  def token_by_user?(a_token)
+    list_tokens = []
```

Comment:
This is the `token_for_text` method! Replace with code from the api:

self.auth_tokens.select { |token| token.authentication_token == a_token }.first

Reply:
Rename this as `token_for_text`